rstfinder/rst_parse.py

```diff
@@ -95,6 +113,132 @@ def segment_and_parse(doc_dict, syntax_parser, segmenter, rst_parser):
     return edu_tokens, rst_parse_trees
 
 
+def from_constituency_trees(tree_strings, segmenter, rst_parser):
+    """
+    Segment and RST parse the document as represented by its constituency trees.
+
+    This function performs discourse segmentation and RST parsing, given a list
+    of constituency trees as strings. All other necessary prerequisites for
+    discourse segmentation and discourse parsing are computed from the given
+    constituency trees.
+
+    The trees _must_ be computed from strings that have the same bracket
+    normalization applied as the texts in the Penn Treebank, i.e., '(' and ')
+    should have been converted to' "-LRB-" and "-RRB-" respectively, '[' and ']'
+    to '-LSB-' and "-RSB-" respectively, and '{' and '}' to "-LCB-" and "-RCB-"
+    respectively. Any paragraph boundaries must be indicated as empty strings
+    and included in the list. For example, if your original document looks like
+    this:
+
+        S1 S2 S3
+
+        S4 S5
+
+        S6 S7 S8 S9
+
+    where "S" represents a sentence, then you have 3 paragraphs, and the list
+    of tree strings should look like this:
+
+        [P1, P2 P3, "", P4, P5, "", P6, P7, P8, P9]
+
+    where "P<N>" represents the parse tree for the sentence "S<N>".
+
+    If there are no empty strings, all sentences will be assumed to be in a
+    single paragraph which will likely yield an incorrect RST parse.
```

Comment:
I don't think this is necessarily true of the RST parser. I think for our later discourse classification, the paragraph boundaries are absolutely necessary. But I think it's perfectly reasonable to get RST parses for a single paragraph.

Reply:
That's fair enough. Will fix!